practical-07/src/store/actions.js

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import route from "../route/route";
+
+export default {
+    fetchData({ commit }) {
+        axios
+            .get("https://testapi.io/api/dartya/resource/cardata")
+            .then((res) => {
+                commit("GET_CAR_DATA", res.data.data);
+            })
+            .catch((error) => {
+                alert(error);
+                console.log(error);
+            });
+    },
+
+    deleteCarDetails({ dispatch }, id) {
+        axios
+            .delete(`https://testapi.io/api/dartya/resource/cardata/${id}`)
+            .then((response) => {
+                if (response.status === 204) {
+                    dispatch("fetchData");
+                }
+            })
+            .catch(() => {
+                alert("can't delete at this moment");
+            });
+    },
+
+    addData({ dispatch }, data) {
+        axios
+            .post("https://testapi.io/api/dartya/resource/cardata", data)
+            .then((res) => {
+                if (res.status === 201) {
+                    dispatch("fetchData");
+                }
+            })
+            .catch(() => {
+                alert("Something went wrong please try again");
+            });
+    },
+
+    editData({ dispatch }, carItem) {
+        axios
+            .put(
+                `https://testapi.io/api/dartya/resource/cardata/${carItem.id}`,
+                carItem
+            )
+            .then((response) => {
+                if (response.status === 200) {
+                    dispatch("fetchData");
+                }
+            })
+            .catch(() => {
+                alert(`cannot update at this moment`);
```

Comment:
Codacy has a fix for the issue: [Replace `················` with `········`](https://app.codacy.com/gh/mansinakrani/vuejs-practicals/pullRequest?prid=9375994)
```suggestion
        alert(`cannot update at this moment`);
```